domain/resource.go

```diff
@@ -17,6 +17,7 @@ type Resource struct {
 	IsDeleted    bool                   `json:"is_deleted,omitempty" yaml:"is_deleted,omitempty"`
 	ParentID     *string                `json:"parent_id,omitempty" yaml:"parent_id,omitempty"`
 	Children     []*Resource            `json:"children,omitempty" yaml:"children,omitempty"`
+	GlobalURN    string                 `json:"global_urn" yaml:"global_urn"`
 }
 
```

Comment:
Can you create a new function here something like below and use it wherever we construct global urn value
```go
// GetGlobalURN returns a GlobalURN for a given source, scope, kind and identifier
// format is urn:<source>:<scope>:<kind>:<identifier>
func GetGlobalURN(source, scope, kind, identifier string) string {
	return fmt.Sprintf("urn:%s:%s:%s:%s", source, scope, kind, identifier)
}
```